tests/ui/if_let_some_result.stderr

```diff
@@ -0,0 +1,31 @@
+error: Matching on `Some` with `ok()` is redundant
+  --> $DIR/if_let_some_result.rs:6:5
+   |
+LL | /     if let Some(y) = x.parse().ok() {
```

Comment:
I meant to reduce the span of the lint so the span should just be `if let Some(y) = x.parse().ok()` in this case instead of the entire `if/else` block as we see here.

Reply:
Yeah, indeed we can reduce output in your suggesting way. But I'm negative to use a hacky span because it may point wrong place. The current lint span tells our intention enough.

Reply:
Sounds good to me. Thanks!